TLSharp.Core/Network/TcpTransport.cs

```diff
@@ -10,103 +12,64 @@ namespace TLSharp.Core.Network
 
     public class TcpTransport : IDisposable
     {
-        private readonly TcpClient tcpClient;
-        private readonly NetworkStream stream;
+        private readonly TLClient tcpClient;
+        private readonly BlockingCollection<TcpMessage> ReceievedMessage = new BlockingCollection<TcpMessage>();
+        //private readonly NetworkStream stream;
         private int sendCounter = 0;
 
         public TcpTransport(string address, int port, TcpClientConnectionHandler handler = null)
         {
-            if (handler == null)
-            {
-                var ipAddress = IPAddress.Parse(address);
-                var endpoint = new IPEndPoint(ipAddress, port);
```

Comment:
As far as i know this library does not handle handlers and proxy so we should work on an implementation later.